### ISSUE TITLE
GraphQLDataSourceRequestKind should be exported

### ISSUE THREAD

**Issue by johgoe** [apollographql/federation#963]
To check if GraphQLDataSourceProcessOptions is of kind INCOMING_OPERATION or of type HEALTH_CHECK or LOADING_SCHEMA the enum GraphQLDataSourceProcessOptions should be exported for other modules. Please add the export to `datasource\index.ts`

**Comment by glasser** [comment]
I agree that would be helpful. In the meantime you can just use the string constants like `'incoming operation`' yourselves, and TypeScript does a good job of raising errors if you use an invalid value.

**Comment by hungcheng-houzz** [comment]
Actually looks like typescript would complain about it.
![image](https://user-images.githubusercontent.com/86639050/156730663-afe46a73-d060-4c94-9720-b042f52ef3f3.png)

**Comment by glasser** [comment]
Ah you're right.  @trevor-scheer we should consider either exporting it as requested here, or replacing the `enum` with a `type X = 'a' | 'b' | 'c'` thing. @brainkim has convinced me that the latter is generally better in TS.
